app/Services/Model/Strategies/PromoCodes/RegularTicketTypePromoCodeValidationStrategy.php

```diff
@@ -0,0 +1,29 @@
+<?php namespace App\Services\Model\Strategies\PromoCodes;
+
+/**
+ * Copyright 2024 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+use App\Models\Foundation\Summit\Registration\PromoCodes\PromoCodesUtils;
+use models\summit\SummitRegistrationPromoCode;
+
+/**
+ * Class RegularTicketTypePromoCodeValidationStrategy
+ * @package App\Services\Model\Strategies\PromoCodes
+ */
+final class RegularTicketTypePromoCodeValidationStrategy implements IPromoCodeValidationStrategy
+{
+    public function isValid(SummitRegistrationPromoCode $promo_code): bool
```

Comment:
here u need also the ticket type and the qty
and u need to check this
https://github.com/OpenStackweb/summit-api/blob/570b8d9ce02913a3cbd789d15c3d23eaf7b63eb9/app/Services/Model/Imp/SummitOrderService.php#L761C8-L761C74
for regular
     $promo_code->validate($owner_email, $owner_company_name);
and then
   if (!$promo_code->canBeAppliedTo($ticket_type)) {
                        Log::debug(sprintf("Promo code %s can not be applied to ticket type %s", $promo_code->getCode(), $ticket_type->getName()));
                        throw new ValidationException(sprintf("Promo code %s can not be applied to Ticket Type %s.", $promo_code->getCode(), $ticket_type->getName()));
                    }
and then check i can be used qty times